### PR TITLE
mds: cleanup nested authpins

### PR DESCRIPTION
**Pull request by ukernel**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by batrick** [comment]
@ukernel there are some merge commits in this PR

**Comment by batrick** [comment]
@ukernel the revert is missing a signed-off-by.

File: src/mds/CDir.cc [hunk 1]
```diff
@@ -2864,12 +2822,56 @@ void CDir::verify_fragstat()
...
+
+  vector<CDir*> dfv;
+  while (!dfq.empty()) {
+    CDir *dir = dfq.front();
+    dfq.pop_front();
+
+    for (auto& p : *dir) {
+      CDentry *dn = p.second;
+      if (!dn->get_linkage()->is_primary())
+	continue;
```
[the remark this answers is not in the document]
Comment (ukernel) [reply src/mds/CDir.cc]: 6 whitespace before 'if'. a tab before 'continue'

[the remark this answers is not in the document]
Comment (batrick) [reply src/mds/CDir.cc]: Sorry, I thought we had a space only rule but according to `CodingStyle`, that's not the case. Carry on.

**Pull request merged**